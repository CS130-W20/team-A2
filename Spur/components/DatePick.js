import React, {useState} from 'react';
import {View, Button, Platform, StyleSheet} from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';

const DatePick = (props) => {
  const [date, setDate] = useState(new Date(1598051730000));
  const [mode, setMode] = useState(props.type);
  const [show, setShow] = useState(false);

  const showMode = currentMode => {
    setShow(true);
    setMode(currentMode);
  };

  const showDatepicker = () => {
    showMode(props.type);
    setShow(!show);
    props.onChange(date);
  };

  const showTimepicker = () => {
    showMode('time');
  };

  const onChange = (event, selectedDate) => {

    console.log(selectedDate);

    props.onChange(selectedDate);
    const currentDate = selectedDate || date;
    
    setDate(currentDate);
    setShow(Platform.OS === 'ios' ? true : false);

    
  };

  return (
    <View>
      <View>
        <Button onPress={showDatepicker} title={props.text} />
      </View>
      {show && (
        <DateTimePicker
          testID="dateTimePicker"
          timeZoneOffsetInMinutes={-7 * 60}
          value={date}
          mode={mode}
          is24Hour={false}
          display="default"
          onChange={onChange}
        />
      )}
    </View>
  );
};

export default DatePick;