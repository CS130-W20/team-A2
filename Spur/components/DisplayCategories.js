import {
    ButtonGroup} from 'react-native-elements'; 
import {
    Text} from 'react-native'; 
import React, { Component } from 'react';
import { CATEGORIES } from '../constants/categories'; 

export default class DisplayCategories extends Component<Props>
{
    constructor(props) {
        super(props); 
        this.state = {
            selectedIndex : 0,
            interests: this.props.interests
        }
        this.updateIndex = this.updateIndex.bind(this)
    }

    /**
     * Function that handles button presses for tags, which is nothing
     * @param {Int} - Index of the button pressed 
     */
    updateIndex (selectedIndex) {
        this.setState({selectedIndex: selectedIndex})
    }

    /**
     * Render DisplayCategories which displays the list of categories
     */
    render() {
        console.log("Print categories")
        var categories = CATEGORIES 
        var catMap = new Map() 
        categories.forEach(function(category) {
            catMap.set(category.id, category.name)
            category.children.forEach(function(child) {
                catMap.set(child.id, child.name)
            })
        })
        console.log(catMap.get(0))
        const buttons = this.state.interests.map(id => (
            <Text>{catMap.get(id)}</Text>
        ))
        const {selectedIndex} = this.state.selectedIndex
        return ( 
            <ButtonGroup
                onPress={this.updateIndex}
                selectedIndex={selectedIndex}
                buttons={buttons}
            />
        )
        /*
        return ( 
            <Text> Testing </Text>
        )
        */
        /*
        const comps = this.props.categories.map(category => (
            <Text> {category.name} </Text>
        ))
        const buttons = comps.map(comp => (
            {element: comp}
        ))
        const {selectedIndex} = this.state

        return ( 
            <ButtonGroup
                onPress={this.updateIndex}
                selectedIndex={selectedIndex}
                buttons={buttons}
                containerStyle={{height: 100}}
            />
        )
        */
    }
}