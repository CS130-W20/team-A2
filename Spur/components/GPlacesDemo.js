import React, { Component } from 'react';
import RNGooglePlaces from 'react-native-google-places';
import {
    StyleSheet,
    Text,
    View,
    ScrollView,
    Dimensions,
    Alert,
    TouchableOpacity } from 'react-native';

export default class GPlacesDemo extends Component {
    openSearchModal() {
      RNGooglePlaces.openAutocompleteModal()
      .then((place) => {
          console.log(place);
          // place represents user's selection from the
          // suggestions and it is a simplified Google Place object.
      })
      .catch(error => console.log(error.message));  // error is a Javascript Error object
    }
   
    render() {
      return (
        <View style={styles.container}>
          <TouchableOpacity
            style={styles.button}
            onPress={() => this.openSearchModal()}
          >
            <Text>Pick a Place</Text>
          </TouchableOpacity>
        </View>
      );
    }
  }

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#fff',
    },
    map: {
      height: 400,
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto',
      justifyContent: 'flex-end',
      alignItems: 'center',	
    },
    inputContainer: {
      borderWidth: 1,
      borderColor: 'lightgrey',
      height: 50,
    },
    input: {
      height: 30,
      backgroundColor: 'lightgrey',
      paddingLeft: 15,
      paddingRight: 15,	
    },
    nameText: {
      height: 30,
      width: 100,
      backgroundColor: 'lightgrey',
      paddingLeft: 15,
      paddingRight: 15,	
    },
    formText: {
      color: 'black',
      fontSize: 20,
      textAlign: 'center',
      margin: 5,
    },
    contentContainer: {
      paddingTop: 30,
    },
    descriptionText: {
      height: 100,
      width: 300,
      backgroundColor: 'lightgrey',
      paddingLeft: 15,
      paddingRight: 15  
    },
    textContainer: {
      flexGrow: 1,
      justifyContent: 'center',
      flexDirection: 'row',
      padding: 5,
    },
  });
  