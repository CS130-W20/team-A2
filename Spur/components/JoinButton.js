import React, {Component} from 'react';
import Icon from 'react-native-vector-icons/FontAwesome';
import DatabaseManager from '../classes/DatabaseManager'; 
import { Image,
	 Platform,
	 StyleSheet,
	 Text,
	 TextInput,
	 TouchableOpacity,
     Alert } from 'react-native';
import {Button} from 'react-native-elements';


export default class JoinButton extends Component<Props>
{
    constructor(props) {
		super(props); 
		//Setup firebase via a databaseManager
		this.databaseManager = new DatabaseManager();
		this.state = {
            event: this.props.event,
            isAttendee: this.props.isAttendee,
            isCheckedIn: this.props.isCheckedIn,
        }
        
    }
    
    /** React render function **/
    render() {
        
        this.databaseManager = new DatabaseManager();
        const eventId = this.props.eventId;
        const event = this.state.event;
        const uid = this.props.uid;
        const upcoming = this.props.upcoming;

        if (!this.state.isAttendee) {
            return( <Button
                title = 'Join Event!'
                onPress={() => 
                    {
                        
                        if (event.attendees.indexOf(uid) < 0) {
                            event.attendees.push(uid);

                            if (upcoming.indexOf(eventId) < 0) {
                                upcoming.push(eventId);
                            }
                            this.databaseManager.updateEvent(eventId, event);
                            this.databaseManager.updateUser(uid, {upcoming: upcoming});
                        }
                    
                        Alert.alert('Event Joined!');
                        
                        this.setState({isAttendee: true, isCheckedIn: false});
                    }
                }
            />);
        } else {

            if (!this.state.isCheckedIn) {
                return( <Button
                    title = 'Check into Event!'
                    onPress={() => 
                        {
                            const checkedIn = event.checked_in ? event.checked_in : []

                            if (checkedIn.indexOf(uid) < 0) {
                                checkedIn.push(uid);
                                this.databaseManager.updateEvent(eventId, {checked_in: checkedIn});
                            }
                           
                            
                            //Alert.alert('Will check user into the event');
                            this.setState({isAttendee: true, isCheckedIn: true});
                        }
                    }
                />);
            } else {
                return( <Button
                    title = "You're checked in!"
                    icon={
                        <Icon
                          name="check"
                          size={15}
                          color="white"
                        />
                      }
                    color="green"
                />);
            }
        }
    }


}
