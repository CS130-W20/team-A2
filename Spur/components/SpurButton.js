import React, {Component} from 'react';
import { Image,
	 Platform,
	 StyleSheet,
	 Text,
	 TextInput,
	 TouchableOpacity,
	 Button,
	 View } from 'react-native';

/** A button class with nice styling **/
export default class SpurButton extends Component<Props>
{
    /** React render function **/
    render() {
	return (
		<View style={{alignSelf: 'center'}}>
			<View style={styles}>
				<Button onPress={this.props.onPress} title={this.props.title} color="#96CA92" />
			</View>
		</View>
  	);
    }
}

const styles = {
	//backgroundColor: "#96CA92",
	flex: 0.2,
	flexDirection: 'row',
	alignItems: 'center',
};
