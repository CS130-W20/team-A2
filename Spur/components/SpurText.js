import React, {Component} from 'react';
import { Image,
	 Platform,
	 StyleSheet,
	 Text,
	 TextInput,
	 TouchableOpacity,
	 View } from 'react-native';
/** Text component with nice styling **/
export default class SpurText extends Component<Props>
{
    /** React render function **/
    render() {
	// copy over style parameters passed down from a parent
	const styleDict = Object.assign({}, styles, this.props.styles);
	return (
		<Text style={styleDict}>{this.props.children}</Text>
  	);
    }
}

const styles = {
	fontSize: 28,
	fontWeight: 'bold',
	color: '#859a80',
};
