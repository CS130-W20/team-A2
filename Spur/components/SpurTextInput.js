import React, {Component} from 'react';
import { Image,
	 Platform,
	 StyleSheet,
	 Text,
	 TextInput,
	 TouchableOpacity,
	 View } from 'react-native';

export default class SpurTextInput extends Component<Props>
{

    render() {
	return (
		<TextInput>Create Event</TextInput>
  	);
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  contentContainer: {
    paddingTop: 30,
  },
  title: {
    fontSize: 19,
    fontWeight: 'bold',
  },
});
