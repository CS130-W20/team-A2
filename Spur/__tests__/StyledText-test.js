import * as React from 'react';
import renderer from 'react-test-renderer';

import { MonoText } from '../components/StyledText';


describe(`Renders Correctly Tester`, () => {
  test('Renders Correctly', () => {
  const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON();
  expect(tree).toMatchSnapshot();
  });
});
