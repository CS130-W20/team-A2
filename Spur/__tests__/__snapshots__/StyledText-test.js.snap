// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly Tester 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`Renders Correctly Tester Renders Correctly 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;
